src/App.vue

```diff
@@ -507,90 +462,120 @@ export default {
-      switch (filing.header.status) {
-        case 'DRAFT':
-        case 'PENDING':
-          // this is a Draft Incorporation Application
-          this.setEntityStatus(EntityStatus.DRAFT_INCORP_APP)
+      // (look for old or new property name)
+      this.localNrNumber = nameRequest?.nrNum || nameRequest?.nrNumber || null
 
-          this.setTasks([
-            {
-              enabled: true,
-              order: 1,
-              task: {
-                filing
-              }
-            }
-          ])
-          break
+      // store Legal Name if present
+      this.setEntityName(nameRequest?.legalName || null)
 
-        case 'COMPLETED':
-        case 'PAID':
-          const incorporationApplication = filing.incorporationApplication
-          if (!incorporationApplication) {
-            throw new Error('Invalid incorporation application object')
-          }
+      switch (filing.status as FilingStatus) {
+        case FilingStatus.DRAFT:
+        case FilingStatus.PENDING:
+          // this is a draft IA
+          this.setEntityStatus(EntityStatus.DRAFT_INCORP_APP)
+          this.storeDraftIa(filing)
+          break
 
-          // this is a paid or completed Incorporation Application
+        case FilingStatus.COMPLETED:
+        case FilingStatus.PAID:
+          // this is a filed IA
           this.setEntityStatus(EntityStatus.FILED_INCORP_APP)
-
-          // set temporary addresses and directors
-          this.storeAddresses({ data: incorporationApplication.offices })
-          const directors = incorporationApplication.parties.filter((party) => party.roles.filter((role) =>
-            role.roleType === 'Director').length !== 0)
-          this.storeDirectors({ data: { directors: [...directors] } })
-
-          this.setFilings([{ filing }])
+          this.storeFiledIa(filing)
           break
 
         default:
           throw new Error('Invalid incorporation application status')
       }
     },
 
-    /** Gets NR data from Legal API. */
-    getNameRequest (): Promise<any> {
-      const url = `nameRequests/${this.localNrNumber}`
-      return axios.get(url)
-        // workaround because data is at "response.data.data"
-        .then(response => Promise.resolve(response.data))
+    /** Stores filing as a task in the Todo List. */
+    storeDraftIa (filing: TaskTodoIF): void {
+      // *** TODO: verify object type for "filing"
+      const taskItem: ApiTaskIF = {
+        enabled: true,
+        order: 1,
+        task: { filing }
+      }
+      this.setTasks([taskItem])
+    },
+
+    storeFiledIa (filing: any): void {
+      // NB: this was already validated in storeIncorpApp()
+      const incorporationApplication = filing.data.incorporationApplication
+
+      // set temporary addresses
+      this.storeAddresses({ data: incorporationApplication.offices || [] })
+
+      // set temporary directors
+      const directors = incorporationApplication.parties?.filter(party => party.roles.filter(role =>
```

Comment:
readability a bit difficult for this.  not a big thing but just something i thought when reading this code

Reply:
Agreed. Is this better? (If not then I'll break it up.)
```
      const directors = incorporationApplication.parties?.filter(party =>
        party.roles.filter(role =>
          role.roleType === Roles.DIRECTOR
        ).length !== 0
      )
```

Reply:
looks good!